internal/github.com/efficientgo/core/backoff/backoff.go

```diff
@@ -0,0 +1,114 @@
+// Copyright (c) The EfficientGo Authors.
+// Licensed under the Apache License 2.0.
+
+// Initially copied from Cortex project.
+
+// Package backoff implements backoff timers which increases wait time on every retry, incredibly useful
+// in distributed system timeout functionalities.
```

Comment:
should we be pulling this via `go.mod` as opposed to having copies of the files? or is this the way we include dependencies for the client library?

Reply:
We try as hard as we can to avoid dependencies. I'm in favor of copying other libraries code (respecting copyright) if possible